### ISSUE TITLE
Cert-controller failing healthcheck - unexpected crd conversion webhook config

### ISSUE THREAD

**Issue by slopezz** [external-secrets/external-secrets#2908]
**Describe the bug**

We have a few external-secrets operator instances deployed through multiple clusters runing different Openshift/k8s versions, and running ESO versions.

Today we have seen that cert-controller pod on a given specific cluster running` Openshift 4.13` (`k8s 1.26`)  running `ESO:v0.9.9` 
 started to not pass the readiness healthcheck giving error `unexpected crd conversion webhook config`, but we don't understand the reason.

Pod logs:
```json
{"level":"error","ts":1701109276.7769604,"logger":"controllers.webhook-certs-updater","msg":"failed to inject conversion webhook","CustomResourceDefinition":{"name":"externalsecrets.external-secrets.io"},"error":"unexpected crd conversion webhook config","stacktrace":"github.com/external-secrets/external-secrets/pkg/controllers/crds.(*Reconciler).Reconcile\n\t/home/runner/work/external-secrets/external-secrets/pkg/controllers/crds/crds_controller.go:123\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/...

{"level":"error","ts":1701109276.7770164,"msg":"Reconciler error","controller":"customresourcedefinition","controllerGroup":"apiextensions.k8s.io","controllerKind":"CustomResourceDefinition","CustomResourceDefinition":{"name":"externalsecrets.external-secrets.io"},"namespace":"","name":"externalsecrets.external-secrets.io","reconcileID":"5f7125bb-2b0b-4636-9601-e01c94c705b6","error":"unexpected crd conversion webhook config","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.g...

{"level":"info","ts":1701109278.931986,"logger":"controller-runtime.healthz","msg":"healthz check failed","statuses":[{},{}]}

{"level":"info","ts":1701109283.8315306,"logger":"controller-runtime.healthz","msg":"healthz check failed","statuses":[{},{}]}
```

However ESO keeps working despite this cert-controller pod to not pass the healthcheck.

**To Reproduce**

We don't know how to reproduce it.

What I can share is that we are running version `0.9.9` on Openshift 4.13 (k8s 1.26) 

External-secrets operator configuration:
```yaml
apiVersion: operator.external-secrets.io/v1alpha1
kind: OperatorConfig
metadata:
  name: cluster
  namespace: external-secrets
spec:
  certController:
    create: true
  installCRDs: false
  prometheus:
    enabled: true
    service:
      port: 8080
  resources:
    limits:
      cpu: 100m
      memory: 256Mi
    requests:
      cpu: 10m
      memory: 96Mi
  webhook:
    create: true
```

**Expected behavior**

Pod passing healthcheck

**Screenshots**
N/A

**Additional context**
Maybe related to https://github.com/external-secrets/external-secrets/pull/2857 ? This PR changed readiness healthcheck and was included in `0.9.9`-

**Comment by moolen** [comment]
The CRDs should be shipped with:

**Comment by slopezz** [comment]
Hi @moolen , thanks for your quick reply, your message help me to find the root cause.

As you mentioned, the ExternalSecret CRD from that specific had conversion empty:

```
  conversion:
    strategy: None
```

Unlike all the others clusters where we install ESO, where conversion webhook was correctly configured.

We never manage CRDs, that's the reason why we contributed to ESO with an OLM installation with the new `OperatorConfig` custom resource, because we trust on OLM to manage operator lifeycle.

Then I found that by accident, yesterday a battery of tests that always run on GitHub actions, that install external dependencies CRDs like external-secrets one for testing purpose, was executed against that specifc cluster, so cluster CRD was wrongly updated with a CRD that had no conversion webhook.

I have just uninstalled the operator, and let OLM to install it again with the correct CRD, and everything is working fine now.

Thanks for your help and sorry for the noise.
